README.md

```diff
@@ -16,6 +16,73 @@ be reused on non-Rock environments provided that you also install its dependenci
 
 The protocol documentation is based on the AceINNA Python library
 
+## Magnetic Calibration
+
+The IMU has a built-in 2D hard/soft iron correction parameters. The E4 message
+outputs the corrected measurements.
+
+To calibrate the mags, first set the parameters to zero. You need to save the
+config and reset the unit to apply.
+
+```
+imu_aceinna_openimu_ctl URL reset # to avoid saving temporary configuration changes
+imu_aceinna_openimu_ctl URL set hard-iron-x 0
+imu_aceinna_openimu_ctl URL set hard-iron-y 0
+imu_aceinna_openimu_ctl URL set soft-iron-angle 0
+imu_aceinna_openimu_ctl URL set soft-iron-ratio 1
+imu_aceinna_openimu_ctl URL save-config
+imu_aceinna_openimu_ctl URL reset # to apply the changes
+```
+
+Then do a few full point turns, as far away as possible from buildings and other
+possible sources of disturbane (AND do not move too much in X/Y to avoid being
+affected by nearby iron). Log the magnetic field information reported by
+the IMU.
+
+Then, you have to fit an ellipse. Assuming you have a Rock system using the
+imu_aceinna_openimu component based on this driver, convert first the
+`magnetic_info` stream to CSV with
+
+```
+pocolog LOGFILE -s TASKNAME.magnetic_info --fields magnetometers --csv > uncorrected_mags.txt
+```
+
+and then in R, source the magnetic_calibration.r script and do the fit with
+
+```
+fit <- magcal("uncorrected_mags.txt")
+```
+
+To visualize the raw data and the fit, now do
+
+```
+plot(fit$raw_x, fit$raw_y)
+lines(fit$fitpoints, col="blue")
+```
+
+To get the ellipse parameters,
+
+```
+fit$fit
+```
+
+### IMU Configuration
+
+The R script returns
+- center in X and Y, which can be provided as-is to the configuration parameters.
+  Note that this is possible because the scripts are written for an IMU mounted
+  upwards, as the motion box is.
+- soft iron angle can be provided as-is
+- soft iron ratio is major / minor
```

Comment:
Does the center parameters you mentioned correspond to the `hard-iron` parameters?